src/charm.py

```diff
@@ -353,6 +403,41 @@ def _get_tls_config(self) -> dict:
             }
         }
 
+    def _get_tracing_config(self) -> dict:
+        """Return dictionary with opentelemetry configuration if available."""
+        # ref: https://doc.traefik.io/traefik/ma<woke forced me to add this>ster/observability/tracing/opentelemetry/
+        if not self._is_tracing_enabled():
+            logger.error("tracing not enabled")
+            return {}
+
+        # traefik supports http and grpc
+        if addr := self._tracing.otlp_grpc_endpoint:
+            grpc = True
+        elif addr := self._tracing.otlp_http_endpoint:
+            grpc = False
+        else:
+            logger.error(
+                "tracing integration is active but none of the "
+                "protocols traefik supports is available."
+            )
+            return {}
+
+        otlp_cfg: Dict[str, Any] = {"address": addr}
+        if self._is_tls_enabled():
+            # todo: we have an option to use CA or to use CERT+KEY authentication.
+            #  what is the difference and which one is better?
+            otlp_cfg["ca"] = _CA_CERT_PATH
+        else:
+            otlp_cfg["insecure"] = True
+
+            # todo: determine when and if we have to use insecureSkipVerify
+            otlp_cfg["tls"] = {"insecureSkipVerify": True}
```

Comment:
What's the diff between `insecureSkipVerify` here and `insecure` above?

Reply:
insecureskipverify = the TLS connection to the [other end] accepts any certificate presented by the server regardless of the hostnames it covers.

Reply:
Great. So what's `insecure` above?

Reply:
accept http. Without insecure=true, it will complain if it cannot connect over https.

Reply:
Had a chat with @simskij and we said we shouldn't have insecureSkipVerify anywhere.